packages/vitest/src/runtime/error.ts

```diff
@@ -2,14 +2,20 @@ import { format } from 'util'
 import { stringify } from '../integrations/chai/jest-matcher-utils'
 
 // https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm
-export function serializeError(val: any): any {
+export function serializeError(val: any, seen = new WeakSet()): any {
   if (!val || typeof val === 'string')
     return val
 
   if (typeof val === 'function')
     return `Function<${val.name}>`
   if (typeof val !== 'object')
     return val
+
+  if (seen.has(val))
```

Comment:
the check should be before the iteration, I'm testing on local and it seems with this approach the memory error still fails

Reply:
With the test or another tkind of error?

I'm still fixing a Maximum Call error for another errors.

Reply:
with `test:all`, testing mocks it works, I'll include your test to check it: 
![imagen](https://user-images.githubusercontent.com/6311119/147418081-21500b76-dd92-4f9e-b08a-3bba29c73379.png)

Reply:
your test also working, I see some invalid WeakSet usage error on some test on `test:ci`:

![imagen](https://user-images.githubusercontent.com/6311119/147418173-c718d4be-357c-498c-a78c-defc550ef0c8.png)

![imagen](https://user-images.githubusercontent.com/6311119/147418144-75a42c7a-9671-48e4-bbc4-8cd1d3ea6a4b.png)

Reply:
Sorry but I do not understand you. 

Is throwing an error runing the new test on your machine or we are talking about the pipeline checks?

Reply:
Runing scripts on Linux Mint (ubuntu base)

-  pnpm test:ci -> Success 
- pnpm test:all -> Success

Reply:
And re-executing throws this:

```
test/mocks test: Failed
undefined
/home/jcabrera/Desktop/vitest/test/mocks:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @vitest/test-lit@ test: `vitest`
Exit status 134
test/react-mui test$ vitest
```

Is that  error? Maybe is the same already commented on discord https://discord.com/channels/917386801235247114/917386801235247117/924403266001993788

Reply:
Yes, it is the same. The error about invalid `WeakSet` usage error was when running `pnpm run test:ci`, your test was not included.

I run all tests using `nr ci` or `pnpm run test:ci` or `pnpm run test:all`: fail 5/6 times with the memory error on discord. In some execution, with your changes on `error.ts` I get invalid `WeakSet` usage error; then I move the check before the `forEach` and the invalid usage error disappear.

I also test each project running `pnpm run test` inside each directory: previous screenshots are the result moving the check before the `forEach` on `mocks` and on `core` with your test also included.

Reply:
Can you please tell me your OS and node + pnpm version? Maybe I can reproduce it

Reply:
I have nvm with node 16.13.0 lts on Windows 10 pro with i9900k proccesor, pnpm latest version